cmake/tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -2,6 +2,8 @@
 # See associated tribits/Copyright.txt file for copyright and license! #
 ########################################################################
 
+set(INTERNAL_IS_MODERN FALSE)
```

Comment:
Might want to namespace this as `HDF5_INTERNAL_IS_MODERN` because these modules are included in base project scope.